kaldb/src/main/java/com/slack/kaldb/chunkManager/ChunkCleanerService.java

```diff
@@ -48,7 +48,8 @@ public int deleteStaleChunksPastCutOff(long staleDataCutOffMs) {
 
     List<Chunk<T>> staleChunks = new ArrayList<>();
     for (Chunk<T> chunk : (Iterable<Chunk<T>>) chunkManager.chunkList) {
-      if (chunk.info().getChunkSnapshotTimeEpochMs() <= staleDataCutOffMs) {
+      if (chunk.isSnapshotted()
```

Comment:
It seems like we should be able to simplify this, and keep all of this logic contained to the `ChunkCleanerService` instead of adding it up to the `Chunk` interface. The concept of a snapshot is specific to the `ReadWriteChunk`, so it's a bit awkward to add this to the interface.

We already have to check the the `snapshotTime` against the `staleDataCutoff` - why not just do something like the following?

```
private boolean chunkIsStale(ChunkInfo chunkInfo, long staleDataCutoffMs) {
  return chunkInfo.getChunkSnapshotTimeEpochMs() > 0 && chunkInfo.getChunkSnapshotTimeEpochMs() <= staleDataCutoffMs;
}
```

We shouldn't need to check `readOnly` - if something has been snapshotted it should have a snapshot time populated, else it should be zero.

Reply:
This would work as well. I like this better. Lemme change the PR